pyszuru/__init__.py

```diff
@@ -31,19 +39,43 @@ def getTag(self, id_: str) -> Tag:
 
     def createTag(self, name: str) -> Tag:
         # Get default tag category
-        tag_cats = [
-            x["name"]
-            for x in self._call("GET", ["tag-categories"])["results"]
-            if x["default"]
-        ]
-        assert len(tag_cats) == 1
-        default_cat = tag_cats[0]
+        tag_cats = self._call("GET", ["tag-categories"])["results"]
+        default_cat = [x for x in tag_cats if x["default"]]
+        assert len(default_cat) == 1
+        default_cat = default_cat[0]["name"]
+
         # Create and return tag
         t = Tag(self, {})
         t._json_new = {"names": [name], "category": default_cat}
         t.push()
         return t
 
+    def getPool(self, id_: int) -> Pool:
+        p = Pool(self, {"id": id_})
+        p.pull()
+        return p
+
+    def createPool(
+        self,
+        name: "str | List[str]",
+        category: str = None,
+        posts: List[Post] = None,
+        description: str = None,
+    ) -> Pool:
+        # Get default pool category
+        tag_cats = self._call("GET", ["pool-categories"])["results"]
+        default_cat = [x for x in tag_cats if x["default"]]
+        assert len(default_cat) == 1
+        default_cat = default_cat[0]["name"]
+
+        p = Pool(self, {})
+        p._json_new = {
+            "names": name,
+            "category": default_cat,
+        }
+        p.push()
```

Comment:
This isn't working. It seems it's pushing to `/pools/` (`Pool.get_class_urlparts() = ['pools']`) but you have to push to `/pool/` instead.

Taking a look at `push()`, I guess it needs a version attribute to push to the right URL. It does push to those correctly when editing a Pool, provided the Pool has a version set.

I see version is only returned from the endpoint to fetch an individual pool, but not from the pool search.
So editing pools works if you fetch them with `API.getPool()` but not if you get the pool with `API.search_pools()` at the moment.

Reply:
It just seems to be inconsistent on Szurubooru's end.  

The Tag and Post objects use the same plural endpoint name for both `POST` requests to create and `GET` requests to query/list/search. The singular endpoint name does not have any `POST` endpoints.

For some reason, Pool objects use the singular endpoint name for this.